R/backend-mssql.R

```diff
@@ -96,7 +97,48 @@
       str_c = sql_paste_infix("", "+", function(x) sql_expr(cast(!!x %as% text))),
       # no built in function: https://stackoverflow.com/questions/230138
       str_to_title = sql_not_supported("str_to_title()"),
-      str_sub = sql_str_sub("SUBSTRING", "LEN")
+      str_sub = sql_str_sub("SUBSTRING", "LEN"),
+
+      # lubridate ---------------------------------------------------------------
+      # https://en.wikibooks.org/wiki/SQL_Dialects_Reference/Functions_and_expressions/Date_and_time_functions
+      as_date = sql_cast("DATE"),
+      as_datetime = sql_cast("DATETIME2"),
```

Comment:
Can you provide a comment that describes why this is `DATETIME2`?

Reply:
Sure. I'm not a SQL Server expert, but my understanding is that DATETME2 is the newer preferred alternative to DATETIME.

Reply:
However, `DATETIME2` is only supported in SQL Server 2008 and newer. I don't know what version of SQL Server dbplyr supports. If lack of support for pre-2008 versions is a problem, we can change it to `DATETIME`.

Reply:
A link to a discussion of the differences would be helpful either way. Is SQL server 2008 in widespread use? It is 10 years old.